system-addon/lib/LocalizationFeed.jsm

```diff
@@ -30,7 +30,14 @@ this.LocalizationFeed = class LocalizationFeed {
   }
 
   updateLocale() {
-    let locale = Services.locale.getRequestedLocale() || DEFAULT_LOCALE;
+    // Just take the first element in the result array, as it should be
+    // the best locale
+    let locale = Services.locale.negotiateLanguages(
+      Services.locale.getAppLocalesAsLangTags(), // desired locales
+      Array.from(Object.keys(this.allStrings)), // available locales
```

Comment:
`Object.keys` already returns an array. (We're not getting `Iterator` keys from `Set`/`Map`.)